scripts/sdk-create-pr.sh

```diff
@@ -51,22 +57,35 @@ git clone ${REPO_URL_SSH} ./
 git config user.name "${COMMIT_NAME}"
 git config user.email "${COMMIT_EMAIL}"
 
-# Replace old SDK with new one
-# Removal of pulled data is necessary because the old version may have files
-# that were deleted in the new version
-rm -rf ./*
-cp -a ${work_dir}/sdk_to_push/. ./
+cp -a . ${work_dir}/sdk_backup
 
-# Create PR with new SDK if there are changes
-if [[ "$REPO_BRANCH" != "main" ]]; then
-    git switch -c "$REPO_BRANCH"
-fi
-git add -A
-if git commit -m "$COMMIT_MESSAGE"; then # Commit will fail if it doesn't contain any changes
-    git push origin "$REPO_BRANCH"
-    if [[ "$REPO_BRANCH" != "main" ]]; then
-        gh pr create --title "$PR_TITLE" --body "$PR_BODY" --head "$REPO_BRANCH" --base "main"
+# Create PR for each new SDK module if there are changes
+for service_path in ${work_dir}/sdk_to_push/services/*; do
+    service="${service_path##*/}"
+
+    # Replace old SDK with new one
+    # Removal of pulled data is necessary because the old version may have files
+    # that were deleted in the new version
+    rm -rf ./services/$service/*
+    cp -a ${work_dir}/sdk_to_push/services/$service/. ./services/$service 
+
+    # Check for changes in the specific folder compared to main
+    service_changes=$(git status --porcelain "services/$service")
+
+    if [[ -n "$service_changes" ]]; then 
+        echo "Creating PR for $service"
+        git switch main # This is needed to create a new branch for the service without including the previously committed files
+        branch="$BRANCH_PREFIX/$service"
+        git switch -c "$branch"
+        
+        git add services/${service}/
+        if [ ! -d "${work_dir}/sdk_backup/services/${service}/" ]; then # Check if it is a newly added SDK module
+            go work use -r . # this removes the use directory if the argument directory doesn’t exist, preventing errors if there is more than one new module in the SDK generation
```

Comment:
Not sure I understood this part, what do you mean by `use directory` and `argument directory`?

Reply:
Ah I see, this removes the entry for the service directory of new services from the `use (...)` list in `go.work`. But I'm still not sure why this is needed, then how would the new service be added to `go.work`?

Reply:
I took the description of the command from the go documentation. See [Learn more about workspaces section on the bottom]( https://go.dev/doc/tutorial/workspaces).

The `go work use -r` also adds the entry if not there. I updated the comment